src/common/interfaces/endpoint-impl.interface.ts

```diff
@@ -0,0 +1,103 @@
+import { ApiProperty } from '@nestjs/swagger';
+import {
+  IsString,
+  IsNotEmpty,
+  IsEnum,
+  IsOptional,
+  IsBoolean,
+} from 'class-validator';
+import { EndpointHeaderType } from '../enums/endpointHeaderType.enum';
+import { ReqBody, HeaderType, QueryType } from './endpoint.interface';
+
+export class ReqBodyClass implements ReqBody {
+  @IsNotEmpty()
+  @IsString()
+  @ApiProperty()
+  key: string;
+
+  @ApiProperty({
+    oneOf: [
+      { type: 'string' },
+      { type: 'boolean' },
+      { type: 'number' },
+      { type: 'object' },
+      { type: 'symbol' },
+      { type: 'array' },
+    ],
+  })
+  value: string | Date | boolean | number | object | symbol | Array<any>;
+
+  constructor(key: string, value: ReqBody['value']) {
+    this.key = key;
+    this.value = value;
+  }
+}
+
+export class HeaderTypeClass implements HeaderType {
+  @IsString()
+  @IsNotEmpty()
+  @ApiProperty()
+  name: string;
+
+  @IsNotEmpty()
+  @IsEnum(EndpointHeaderType)
+  @ApiProperty({
+    enum: EndpointHeaderType,
+    enumName: 'EndpointHeaderType',
+  })
+  type: EndpointHeaderType;
+
+  @IsOptional()
+  @ApiProperty({ type: ReqBodyClass })
+  value: ReqBody['value'];
+
+  @IsBoolean()
+  @ApiProperty()
+  required: boolean;
+
+  constructor(
```

Comment:
wow, i havent seen this b4 @devynlab. are you sure its not prone to bugs?

Reply:
I have tested and it works just fine, plus the added validation. 
But of course I'll need you guys to also just validate with your tests...